sklearn/metrics/cluster/supervised.py

```diff
@@ -50,6 +52,21 @@ def check_clusterings(labels_true, labels_pred):
     return labels_true, labels_pred
 
 
+def _generalized_average(U, V, average_method):
+    """Return a particular mean of two numbers."""
+    if average_method == "min":
+        return max(min(U, V), 1e-10)
+    elif average_method == "geometric":
+        return max(np.sqrt(U * V), 1e-10)  # Avoids zero-division error
```

Comment:
... but you'll land up with near-infinite measures anyway?? Maybe we're better off warning the user.

Reply:
perhaps use `eps` defined as `np.finfo('float64').eps` instead of this arbitrary value.

It's also awkward having this comment here and not the first time this eps is used.

Reply:
I think the `normalizer = max(normalizer, eps)` should happen in  NMI, actually, not here. It's not directly applicable in AMI which should check for `normalizer - emi == 0`.

I think in the case that the denominator is 0, we should be issuing at least a warning, whether or not we then return a finite (backwards compatible in NMI) or infinite value.